packages/dashboard/src/components/robots/robot-page.tsx

```diff
@@ -154,6 +154,12 @@ export function RobotPage() {
           });
           newRobotStates[robot.name] = robot;
           robotStatesRef.current[robot.name] = robot;
+          if (selectedRobot && selectedRobot.name && robot.name) {
+            setSelectedRobot(robot);
+            if (robot.task_id) {
+              setSelectedTask(taskStatesRef.current[robot.task_id]);
+            }
+          }
```

Comment:
`selectRobot` is already handled by the click handler, in what cases are the click handlers insufficient?

Reply:
Yes, well the user need to click on a task to see the task-progress get updated.